libs/common/src/interfaces/crypto.interface.ts

```diff
@@ -0,0 +1,53 @@
+import { EncryptService } from "../abstractions/encrypt.service";
+import { SymmetricCryptoKey } from "../models/domain/symmetric-crypto-key";
+
+export function nullableFactory<T extends new (...args: any) => any>(
+  c: T,
+  ...args: ConstructorParameters<T>
+): InstanceType<T> | undefined {
+  if (args[0] == null) {
+    return null;
+  }
+
+  return new c(...(args as any[]));
+}
+
+/**
+ * Domain model that can be decrypted using views.
+ */
+export interface DecryptableDomain {
+  /**
+   * Unique GUID for the key used to encrypt the data
+   */
+  keyIdentifier(): string | null;
+}
+
+/**
+ * View model that encrypts to Domain.
+ */
+export interface Encryptable<Domain> {
```

Comment:
Please prefix type parameters with `T`

```suggestion
export interface Encryptable<TDomain> {
```